python/hail/dataset.py

```diff
@@ -3403,43 +3403,58 @@ def min_rep(self, left_aligned=False):
         return VariantDataset(self.hc, self._jvds.minRep(left_aligned))
 
     @handle_py4j
-    @require_biallelic
     @record_method
-    @typecheck_method(scores=strlike,
-                      loadings=nullable(strlike),
-                      eigenvalues=nullable(strlike),
-                      k=integral,
+    @require_biallelic
+    @typecheck_method(k=integral,
+                      compute_loadings=bool,
                       as_array=bool)
-    def pca(self, scores, loadings=None, eigenvalues=None, k=10, as_array=False):
-        """Run Principal Component Analysis (PCA) on the matrix of genotypes.
+    def normalized_genotype_pca(self, k=10, compute_loadings=False, as_array=False):
+        """Run Principal Component Analysis (PCA) on the HWE-normalized genotype matrix of this VSM.
 
-        .. include:: _templates/req_tvariant.rst
+        Variants with constant or all-missing genotypes will be removed before evaluation.
 
         .. include:: _templates/req_biallelic.rst
 
-        **Examples**
+        :param int k: Number of principal components.
 
-        Compute the top 5 principal component scores, stored as sample annotations ``sa.scores.PC1``, ..., ``sa.scores.PC5`` of type Double:
+        :param bool compute_loadings: If true, computes variant loadings. Default: False
 
-        >>> vds_result = vds.pca('sa.scores', k=5)
+        :param bool as_array: If true, stores KeyTable rows as type Array rather than Struct
 
-        Compute the top 5 principal component scores, loadings, and eigenvalues, stored as annotations ``sa.scores``, ``va.loadings``, and ``global.evals`` of type Array[Double]:
+        :return: Tuple of: list of eigenvalues, KeyTable of scores, KeyTable of loadings
+        :rtype: (list of float, :py:class:`.KeyTable`, :py:class:`.KeyTable`)
+        """
 
-        >>> vds_result = vds.pca('sa.scores', 'va.loadings', 'global.evals', 5, as_array=True)
+        vds = self.annotate_variants_expr('va.mean = gs.map(g => g.GT.gt).sum()/gs.filter(g => g.GT.isDefined).count()')\
```

Comment:
`g.GT.isDefined` should be `isDefined(g.GT)`.  This only works by virtual of some lingering sloppiness in the function registry lookup.

Reply:
I think pretty consistently use spaces around our operators, e.g., /.